server.js

```diff
@@ -33,10 +33,26 @@ app.get('/admin', function(req, res) {
   res.sendFile(__dirname + '/templates/admin.html');
 });
 
+app.get('/bobs', function (req, res) {
+  res.redirect(301, '/api/bobs/');
+});
+
 // Show edit page on /bobs/:bobid
 app.route('/bobs/:bobid')
   .get(function(req, res) {
-    res.sendFile(path.join(__dirname, '/templates/editbob.html'));
+    if(req.params.bobid.length === 24){
+      db.Bob.getOneBob({ _id: db.ObjectId(req.params.bobid)}).then(function success(data) {
+        if(data){
+          res.sendFile(path.join(__dirname, '/templates/editbob.html'));
```

Comment:
hmm, it feels a little redundant to be checking for the bob here and later querying for the bob once again from the frontend. Kinda weird situation as the first thing that comes to mind is using node's `res.render()` to send the bob returned by the query. You would then be able to access the bob data in the template, however this means that to use it in the javascript we have to do something kinda gross, which is create a global variable in the template that is later referred to in the included script file. 

By this I mean:
changing this line to something like `res.render(path.join(__dirname, '/templates/editbob.html'), {bob: data});`

then in editbob.html
```
...
<script>var bob = {{ bob }}</script>
<script src="editbob.js"></script>
...
```

and in editbob.js:
```
...
var cur_bob = window.bob; // I think just plain bob would work but window. is more explicit about pulling it out of magic global land
...
```